### PR TITLE
fix getElevationGain implementation

### PR DESCRIPTION
**Pull request by nickspeal**
I found a bug in Bikesy production, where the getElevationGain function was always returning `0` because of an incorrect assumption about the type of the input variable,`profile`. The role of this function is to count to the total elevation gain along a route, by iterating through the list of path segments, checking if the segment is a net climb, and if so adding it to a tally. But when iterating through the `profile` array, it was trying to access data on each element in the array with index `1`, as if each element was a list. But upon inspection, each element is an object. 

I first changed the data access pattern to use the string key to access the property on an object. I then refactored the function to be more readable.

## Testing

Bikesy.com:

Screen Shot 2021-11-06 at 2 48 20 PM

On this branch:
Screen Shot 2021-11-06 at 2 48 41 PM

Also, I did a test replacing `profile` with an empty list and confirmed that nothing crashed. It just rendered "0 feet of total climbing"

### CODE REVIEW

File: src/lib/helper.js [hunk 1]
```diff
@@ -31,13 +31,15 @@ export function formatTime(miles) {
...
-    if (idx < profile.length - 1 && profile[idx][1] < profile[idx + 1][1]) {
-      totalElevGain += profile[idx + 1][1] - profile[idx][1];
+  for (let idx = 0; idx < (profile.length - 1); idx++) {
+    const elevationStep = profile[idx + 1]['elevation'] - profile[idx]['elevation']
+    if (elevationStep > 0) {
+      totalElevGain += elevationStep
     }
-  });
-
+  }
```
Comment (robotnealan) [inline src/lib/helper.js:42]: What do you think of writing this using `reduce`? I'm a stickler for functional JS that doesn't mutate variables over and over.

```
const totalElevGain = profile.reduce((accum, {elevation}, index) => {
  const elevationStep = profile[index].elevation - elevation;
  if(elevationStep > 0) return accum + elevationStep;
  return accum;
}, 0);
```

Comment (nickspeal) [reply src/lib/helper.js:42]: Less readable IMO. Less obvious that we are comparing the elevation of the next point to the current point.

**Comment by brendannee** [comment]
Great catch, thanks for this PR.
